packages/cli/src/lib/live-server/index.js

```diff
@@ -402,16 +402,23 @@ LiveServer.start = function(options) {
         console.log("CSS change detected".magenta, changePath);
       else console.log("Change detected".cyan, changePath);
     }
-    
-    // CHANGED: Prepare tab entry data before issuing reload
-    LiveServer.activeTabs.forEach(tab => {
-      if (tab.client) {
-        // Clear the client from the entry to be refilled in the socket establishment phase after reload
-        const client = tab.client;
-        tab.client = undefined;
-        client.send(cssChange ? 'refreshcss' : 'reload');
-      }
-    });
+
+    // CHANGED: Modified the send message to clients routine
+
+    if (cssChange) {
+      LiveServer.sendMessageToActiveTabs('refreshcss');
+      return;
+    }
+
+    // Only reload active tabs if the changed file is opened in one of them
+    let normalizedPath = path.relative(root, changePath);
+    if (path.sep !== '/') {
```

Comment:
It's possible, but for that we may need to add a new import to `@markbind/core/src/utils/fsUtil` wrt. the original live-server, which ends up coupling the patch with the `core` package. Though a bit minor in reason, I wanted the patch to be able to stand on its own unless we really need it to interact with our business logic.

I checked out the source of `ensurePosix`, the `ensure-posix-path` package (the package only contains that function). Thought I might add that package as an import instead of the `core` util, but it is pretty much the same as what I did hahaha. I might have reinvented the wheel but it's just a few lines here. What do you think?

Reply:
I think that's ok _for now_, its already rather coupled (e.g. `cli/index.js` heavily uses utils as well). =X

We might resolve this with the different package restructure above https://github.com/MarkBind/markbind/pull/1539#discussion_r615544434, or, as a more long term solution, also extract common utility functions to another package, something like https://github.com/vuejs/vuepress/tree/master/packages/%40vuepress/shared-utils/src

Reply:
Okay then, I will import the utils in the patch for now.

> We might resolve this with the different package restructure above #1539 (comment), or, as a more long term solution, also extract common utility functions to another package, something like https://github.com/vuejs/vuepress/tree/master/packages/%40vuepress/shared-utils/src

But I agree with this, I will post an issue around this possible refactoring.